### PR TITLE
Add Annotation to Control Inline List Conversion in Kustomize Resources"

### PR DESCRIPTION
**Pull request by isarns**
This PR introduces a new annotation to Kustomize resources: `kustomize.config.k8s.io/convertToInlineList`. When set to `false`, this annotation prevents objects with names ending in "List" from being converted to inline resources, allowing them to be treated at face value.

### Changes:
- Added the `kustomize.config.k8s.io/convertToInlineList: false` annotation.
- Updated the handling logic to respect this annotation and avoid converting specified lists into inline resources.

### Example:
For example, the resource definition [ManagedPrefixList](https://marketplace.upbound.io/providers/upbound/provider-aws-ec2/v1.8.0/resources/ec2.aws.upbound.io/ManagedPrefixList/v1beta1) does not require conversion to an inline list, and with this annotation, Kustomize will respect its original format.

### Issues Addressed:
- Fixes issue #5042
- Fixes issue #5485
- Heavily inspired by issue #5045

### Feedback Request:
As someone new to Golang, I understand that this PR might not be perfect. I welcome any guidance, suggestions, or improvements from the community to refine this feature further.

### CODE REVIEW

**Comment by koba1t** [comment]
Hi @isarns 
Thanks for your contribution!

I read the issue, but I think we should try another approach from this PR.
I think the current problem is that kustomize recognizes it as a `List` resource for all resources with a name of the `*List`.
ref: https://github.com/kubernetes-sigs/kustomize/issues/5042#issuecomment-1518497660

I believe the better way is the [cailynse](https://github.com/cailynse)'s approach that changes to recognize `*List`. 
Due to
https://github.com/kubernetes-sigs/kustomize/pull/5045
https://github.com/kubernetes-sigs/kustomize/blob/cc9dd34216d98be8efe341939094867b553b2cbc/api/resource/factory.go#L193

FYI: We already have API schemes on [here](https://github.com/kubernetes-sigs/kustomize/pull/5745).

Could you change to use the above approach to this PR?
I appreciate you taking this Problem!

**Comment by isarns** [comment]
@koba1t Thanks for the reply. 

I have refactored my PR. 
Added another test that take a "Real" list object and check that kustomize create the individual resources from it.

File: api/resource/factory.go [hunk 1]
```diff
@@ -213,7 +202,10 @@ func (rf *Factory) inlineAnyEmbeddedLists(
 		}
 		items, ok := m["items"]
```
Comment (koba1t) [inline api/resource/factory.go:203]: I worry about what happens when an empty PodList is put.

```
apiVersion: v1
kind: PodList
items: []
```

Can you add that test case?

**Review by koba1t — CHANGES_REQUESTED** [review]
@isarns 
Thanks for fixing your PR!
I added one comment. Could you check that?

**Comment by isarns** [comment]
@koba1t Thanks. 

I checked what current kustomize does with an empty “real” list, and it returns an empty response (No resource).
The same happens with my PR.

**Comment by chkp-tron** [comment]
Really looking forward to this

**Comment by isarns** [comment]
Hey @koba1t. 

Can you have another look, please?

**Comment by koba1t** [comment]
@isarns 
Thank you for your response. I'll check.

**Comment by koba1t** [comment]
@isarns 
Could you fix lint errors?

**Comment by isarns** [comment]
> Hi @isarns Could you fix lint errors and test failure?

Thanks! Fixed those.

**Comment by koba1t** [comment]
@isarns 
Thanks for your contribution!

/lgtm
/approve

**Pull request merged**